### DISCUSSION TITLE
How about some documentation on using this as a library?

### DISCUSSION THREAD

**Discussion by frankhaugen** [dotnet/ClangSharp#561] [Ideas]
I get that this is a super-niche tool not unlike xsd.exe, so usability is not a huge priority, but the lack of clearity around how to use the generator nuget as a library, is frustrating.

I am looking into making a roslyn source generator that leverages this, but it seems pretty unclear how to use the code. Even looking at the tests is confusing me more, not less about how I could use this.

It would therefor be nice with a few lines of docs showing how to use this

**Comment by typeseven** [comment]
This might help a bit: https://clang.llvm.org/docs/LibClang.html

**Comment by tannergooding** [comment]
I've added multiple additional docs since this ask and have updated the readme to link to them and to the official libClang docs, since we largely just wrap that for the interop side.
